backoff.go

```diff
@@ -27,12 +27,37 @@ import (
 
 // DefaultBackoffConfig uses values specified for backoff in
 // https://github.com/grpc/grpc/blob/master/doc/connection-backoff.md.
+//
+// Deprecated: use ConnectParams instead.
 var DefaultBackoffConfig = BackoffConfig{
 	MaxDelay: 120 * time.Second,
 }
 
 // BackoffConfig defines the parameters for the default gRPC backoff strategy.
+//
+// Deprecated: use ConnectParams instead.
 type BackoffConfig struct {
 	// MaxDelay is the upper bound of backoff delay.
 	MaxDelay time.Duration
 }
+
+// ConnectParams defines the parameters for connecting and retrying. Users
+// are encouraged to use this instead of BackoffConfig.
+// https://github.com/grpc/grpc/blob/master/doc/connection-backoff.md.
+//
+// This API is EXPERIMENTAL.
+type ConnectParams struct {
+	// BackoffBaseDelay is the amount of time to backoff after the first
```

Comment:
Could we replace all the backoff fields here with a single `BackoffConfig` field, and add these fields to the `BackoffConfig` (excluding `MinConnectTimeout`)?  Theoretically we could pass `BackoffConfig` to `Exponential` directly, though that is unfortunately not easy because it's in a different package and has circular dependency issues.  We'd need a new, non-internal `backoff` package.  This may actually be a good idea.